zanata-war/src/main/java/org/zanata/servlet/MultiFileUploadServlet.java

```diff
@@ -392,7 +392,23 @@ private DocumentFileUploadForm createUploadFormForItem(FileItem item) throws IOE
             form.setFirst(true);
             form.setLast(true);
             form.setSize(item.getSize());
-            form.setFileType(translationFileServiceImpl.extractExtension(item.getName()));
+
+            String extension = FileUtil.extractExtension(item.getName());
+            String fileType = null;
+            if(!fileTypes.isEmpty()) {
+                for(String parsedFileType: fileTypes) {
+                    DocumentType docType = DocumentType.valueOf(parsedFileType);
+                    if(docType != null && docType.getSourceExtensions().contains(extension)) {
+                        fileType = docType.name();
+                        break;
+                    }
+                }
+            }
+            fileType =
+                    fileType != null ? FileUtil
+                            .extractExtension(item.getName()) : fileType;
```

Comment:
- [ ] Could just initialize fileType with the extracted extension and remove this line.

```
String filtType = FileUtil.extractExtension(item.getName());
```